catalogue/webapp/services/prismic/fetch/articles.ts

```diff
@@ -2,32 +2,169 @@ import { Story } from '../types/story';
+              }
+            }
+          }
+          promo {
+            ... on ArticlesPromoEditorialimage {
+              primary {
+                image
+                link
+                caption
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
 
 export type PrismicQueryProps = {
   query: Query;
   pageSize: number;
   type?: string;
 };
 
+// This function will:
+// 1. Call the prismicGraphQLClient function with queryString from query and get nth number of stories by pageSize
+// Then, it will transform the above response into a list of stories in the format of ResultList
+// 2. If the query contains a page number, it will query the prismicGraphQLClient function with the cursor
+// and get the next nth number of stories by pageSize
 export async function getStories({
   query,
   pageSize,
 }: PrismicQueryProps): Promise<PrismicResultsList<Story> | PrismicApiError> {
   try {
-    const res = await prismicGraphQLClient('articles', query, pageSize);
-    const { allArticless } = await res;
-    const { edges } = allArticless;
-    const stories = await transformPrismicResponse(['articles'], edges);
-
-    return {
-      type: 'ResultList',
-      totalResults: allArticless.totalCount,
-      totalPages: Math.ceil(allArticless.totalCount / pageSize),
-      results: stories,
-      pageSize: pageSize,
-      prevPage: null,
-      nextPage: null,
+    // Create a map of all cursors to their respective page number by pageSize
+    // A cursor is a pointer to a node within the returned list of results from a graphql query
+    const fetchAllCursors = async (
+      type: string,
+      query: Query,
+      pageSize: number
+    ) => {
+      const results = await prismicGraphQLClient(type, query, 100);
+      const { allArticless } = await results;
+      const { edges } = allArticless;
+      const getStoriesCursors = edges.map(edge => {
+        return {
+          cursor: edge.cursor,
+        };
+      });
+      // We now get the first and nth cursor for first and nth page based on pageSize
+      const cursorsPaginated = (cursors, pagesize) =>
+        cursors
+          .filter(
+            (cursor, index) => index % pagesize === pagesize - 1 || index === 0
+          )
+          .map((e, index) => {
+            return { cursor: e.cursor, page: index + 1 };
+          });
+
+      return cursorsPaginated(getStoriesCursors, pageSize);
+    };
+
+    // Get the relevant cursor for the page number
+    const currentPageCursor = async (
+      type: string,
+      query: Query,
+      pageSize: number
+    ) => {
+      const allCursors = await fetchAllCursors('articles', query, pageSize);
+      const currentPage = query.page ? query.page : 1;
+      const getCurrentPageCursor = allCursors.find(
+        cursor => cursor.page === parseFloat(String(currentPage))
+      );
+      return getCurrentPageCursor?.cursor;
     };
+
+    const fetchStories = async (
+      type: string,
```

Comment:
Let's explain why this `type` is here.

Reply:
nit: maybe mention it's used on events and exhibitions, and if they're no longer using it we can remove from all three?